pkgs/applications/science/math/matlab/matlab.nix

```diff
@@ -0,0 +1,161 @@
+{
+  lib,
+  stdenvNoCC,
+  buildFHSEnv,
+  makeDesktopItem,
+
+  fetchFromMPM,
+  matlab-package-manager,
+
+  source ? fetchFromMPM {
+    release = "R2024b";
+    update = 1;
+    extra-products = [ ];
+    hash = "sha256-t7+vIk7254mo7dUaRFjZp/qHRpVWnIhy3OEoZfhwqUI=";
+  },
+  withExtraPackages ? pkgs: [ ],
+}:
+
+assert source ? matlab;
+
+let
+  matlab = stdenvNoCC.mkDerivation {
+    pname = "matlab";
+    inherit (source) version;
+
+    dontUnpack = true;
+
+    installPhase = ''
+      # As mpm is bubblewrapped, it cannot write to $out directly
+      # https://github.com/NixOS/nixpkgs/issues/239017
+      ${lib.getExe matlab-package-manager} install --source "${source}" --destination matlab --products ${lib.escapeShellArgs source.matlab.products}
+      mv matlab $out
+      ln -s /etc/matlab $out/licenses
+    '';
+
+    dontFixup = true;
+  };
+
+  desktopItem = makeDesktopItem {
+    name = "matlab";
+    exec = "matlab -desktop -useStartupFolderPref";
+    comment = "MATLAB is a programming and numeric computing platform for engineering and scientific applications.";
+    desktopName = "MATLAB ${source.matlab.release}";
+    categories = [
+      "Development"
+      "Math"
+      "Science"
+    ];
+    icon = "${matlab}/bin/glnxa64/cef_resources/matlab_icon.png";
```

Comment:
Please use an icon name (e.g. "matlab") and not a path, since otherwise it won't allow themes overriding icons. Install/link the icon to $out/share/pixmaps.